### PR TITLE
spatial/r3: small cleanup

### PR DESCRIPTION
**Pull request by vladimir-ch**
- spatial/r3: remove netlib from dependency requirements
- spatial/r3: fix doc comments to start with function name

Please take a look.

### CODE REVIEW

File: spatial/r3/mat.go [hunk 1]
```diff
@@ -187,12 +185,12 @@ func (m *Mat) Outer(alpha float64, x, y Vec) {
 	m.Set(2, 2, az*y.Z)
 }
 
-// Sets the rotation component of this matrix to the rotation specified by q,
-// as outlined at https://en.wikipedia.org/wiki/Rotation_matrix#Quaternion.
+// RotationFromQuat converts the quaternion q to the corresponding orthogonal
+// matrix and stores the result in the receiver. q should be normalized but this
```
Comment (vladimir-ch) [inline spatial/r3/mat.go:189]: Why not normalize the quaternion inside the function?

Comment (kortschak) [reply spatial/r3/mat.go:189]: In the case that the user want a rotation and scale, that becomes impossible.

**Review by kortschak — APPROVED** [review]

File: spatial/r3/mat.go (hunk 1 above)
Comment (vladimir-ch) [reply spatial/r3/mat.go:189]: But then we're in an undocumented territory and the existing documentation and naming becomes inaccurate: the matrix is not orthogonal and so not a rotation.

Comment (kortschak) [reply spatial/r3/mat.go:189]: Fair enough; probably we should do that then and note that it's normalised. Thanks for that.

Comment (vladimir-ch) [reply spatial/r3/mat.go:189]: Yeah, I think if scaling is needed it can be easily achieved with `Scale`.

Comment (vladimir-ch) [reply spatial/r3/mat.go:189]: I've been thinking about this a bit more and it's probably the cleanest to leave it as is. The behavior is documented for unit quaternions, non-unit behavior is not documented and we don't need to test it. We could maybe replace "should" with "must", @kortschak ?

Comment (kortschak) [reply spatial/r3/mat.go:189]: That's fine with me. I think that people should be using quaternions for this kind of thing where all this just works.

**Pull request merged**